etools-refresh-behavior.html

```diff
@@ -0,0 +1,73 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../etools-dexiejs/etools-dexiejs.html">
+<script>
+  window.etoolsBehaviors = window.etoolsBehaviors || {};
+
+  /** @polymerBehavior etoolsBehaviors.EtoolsRefreshBehavior */
+ etoolsBehaviors.EtoolsRefreshBehavior = {
+    properties: {
+      dexieDbsNumber: Number,
+      deleteAttemptsOnDbs: Array,
+      refreshInProgress: Boolean,
+    },
+
+    ready: function() {
+      this.dexieDbsNumber = 0;
+      this.deleteAttemptsOnDbs = [];
+    },
+
+    refresh: function() {
+      this.refreshInProgress = true;
+      //**Important : Do not clear localStorage before Dexie db ,
+      // because for Firefox, IE, Edge and Safari, Dexie stores the db names
+      // in the localStorage under the key Dexie.DatabaseNames
+      // and method Dexie.getDabasesNames searches for this key
+
+      // clear DexieDBs for current host
+      this.clearDexieDbs();
+
+        // clear localStorage data
+      this.clearLocalStorage();
+    },
+
+    clearLocalStorage: function() {
+      localStorage.clear();
+    },
+
+    clearDexieDbs: function() {
+      // except Chrome and Opera this method will delete only the dbs created with Dexie
+      Dexie.getDatabaseNames(function(dbsNames) {
+        this.dexieDbsNumber = dbsNames.length;
+        if (this.dexieDbsNumber > 0) {
+          dbsNames.forEach(function(dbName) {
```

Comment:
@daniel-tabarcea could we refactor this and use `Promise.all` for better check when all dexie dbs were deleted. By the time this was implemented I missed this option that fits better with this functionality. What do you think?